File: app/src/main/java/com/example/ass07/admin/booking/Booking.kt
package com.example.ass07.admin.booking

import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class Booking(

    // ข้อมูลการจอง
    @Expose @SerializedName("booking_id") val bookingId: Int,
    @Expose @SerializedName("check_in") val checkIn: String,
    @Expose @SerializedName("check_out") val checkOut: String,
    @Expose @SerializedName("additional_info") val additionalInfo: String?,
    @Expose @SerializedName("pay") val pay: Int,
    @Expose @SerializedName("adjust") val adjust: Int?,
    @Expose @SerializedName("total_pay") val totalPay: Int,
    @Expose @SerializedName("booking_status") val status: Int,
    @Expose @SerializedName("payment_method") val paymentMethod: Int,
    @Expose @SerializedName("pet_id") val petId: Int,
    @Expose @SerializedName("room_id") val roomId: Int,
    @Expose @SerializedName("created_at") val createdAt: String,
    @Expose @SerializedName("updated_at") val updatedAt: String,
    @Expose @SerializedName("deleted_at") val deletedAt: String?,

    // ข้อมูลสัตว์เลี้ยง
    @Expose @SerializedName("pet_name") val petName: String,
    @Expose @SerializedName("pet_gender") val petGender: String,
    @Expose @SerializedName("pet_breed") val petBreed: String,
    @Expose @SerializedName("pet_age") val petAge: Int,
    @Expose @SerializedName("pet_height") val petHeight: Double,
    @Expose @SerializedName("pet_weight") val petWeight: Double,

    // ข้อมูลผู้ใช้
    @Expose @SerializedName("name") val name: String,
    @Expose @SerializedName("tell_number") val tellNumber: String,
    @Expose @SerializedName("email") val email: String,

    // ข้อมูลห้องพัก
    @Expose @SerializedName("type_type_id") val typeTypeId: Int,
    @Expose @SerializedName("status") val roomStatus: Int,

    // ข้อมูลประเภทห้อง
    @Expose @SerializedName("name_type") val roomType: String,
    @Expose @SerializedName("price_per_day") val pricePerDay: Int,
    @Expose @SerializedName("image") val roomImage: String?,
    @Expose @SerializedName("pet_type") val petType: Int,

    // ข้อมูลประเภทสัตว์ที่เข้าพัก
    @Expose @SerializedName("pet_name_type") val petNameType: String,

    // การชำระเงิน
    @Expose @SerializedName("method_name") val methodName: String

) : Parcelable

File: app/data/routes/booking.js
const express = require("express");
const router = express.Router();
const dbConn = require("../db"); // เชื่อมต่อกับฐานข้อมูล

// ดึงข้อมูลการจองทั้งหมด
router.get("/bookings", function (req, res) {
  const query = `
    SELECT
      bookings.*,
      pets.pet_name, pets.pet_gender, pets.pet_breed, pets.pet_age, pets.pet_height, pets.pet_weight,
      users.name , users.tell_number, users.email,
      rooms.room_id , rooms.type_type_id ,rooms.status,
      room_type.name_type, room_type.price_per_day , room_type.image, room_type.pet_type,
      pet_type.pet_name_type,
      payment_methods.method_name
    FROM bookings
    JOIN pets ON bookings.pet_id = pets.pet_id
    JOIN users ON pets.user_id = users.user_id
    JOIN rooms ON bookings.room_id = rooms.room_id
    JOIN room_type ON rooms.type_type_id = room_type.type_id
    JOIN pet_type ON room_type.pet_type = pet_type.pet_type_id
    JOIN payment_methods ON bookings.payment_method = payment_methods.method_id
    ORDER BY bookings.booking_id ASC`;
  
  dbConn.query(query, function (error, results) {
    if (error) throw error;
    return res.send(results);
  });
});

module.exports = router;
